docs(block-node): add glossary and replace scattered Key Terms sections with inline links (#3142)
hiero-ledger/hiero-block-node: docs/block-node/architecture/architecture-overview.md

- **BlockVerification:** Blocks are verified for integrity using the `VerificationServicePlugin` which build the virtual
  merkle tree and validate the block proof prior to persistence.

## Plugins

BlockNode's functionality is extended through a variety of plugins, each implementing the `BlockNodePlugin` interface.
Key plugins include:

> - **BackfillPlugin:** Guarantees the stored block stream is complete by retrieving missing blocks from other Block Nodes.

Reviewer: Not really a guarantee, more of a best effort.

```suggestion
- **BackfillPlugin:** Helps to ensure the stored block stream is complete by retrieving missing blocks from other Block Nodes.
```

- **BlockAccessServicePlugin:** Provides a block retrieval API.

> - **BlocksFilesHistoricPlugin:** Handles block persistence and retrieval of older persisted blocks.

Reviewer: Not older, just long term

```suggestion
- **BlocksFilesHistoricPlugin:** Provides long term block persistence and retrieval.
```

> - **BlocksFilesRecentPlugin:** Handles block persistence, retrieval, and retention policies of recently streamed blocks.

Reviewer: Short term rather than recent.

```suggestion
- **BlocksFilesRecentPlugin:** Provides short term block persistence and retrieval, with a retention policy to limit storage use and duration.
```

> - **HealthServicePlugin:** Provides kubernetes health check endpoints.

Reviewer: ```suggestion
- **HealthServicePlugin:** Provides kubernetes health check endpoints, additional status endpoints to integrate with Kubernetes features, and overall system health decision support.
```

> - **S3ArchivePlugin:** Archives blocks to S3 API compatible remote storage services.

Reviewer: This plugin is deprecated and removed.

Replaced with `storage-cloud-archive` and `storage-cloud-expanded`.

- **ServerStatusServicePlugin:** Provides block node status API endpoints.
- **StreamPublisherPlugin:** Provides a block stream publishing API as documented in the [communication protocol](./../../design/communication-protocol/README.md).
- **SubscriberServicePlugin:** Provides an _unverified_ Block Subscription API.

> - **VerificationServicePlugin:** Verifies incoming blocks for integrity prior to persistence.

Reviewer: Recent name change:

```suggestion
- **BlockVerificationServicePlugin:** Verifies incoming blocks for integrity prior to persistence.
```

- `stream-publisher`: Plugins for a Stream publishing API.

> - `stream-subscriber`: Plugins for Stream subscribing API.

Reviewer: Each module is a single plugin.

```suggestion
- `stream-publisher`: Plugin for a Stream publishing API.
- `stream-subscriber`: Plugin for Stream subscribing API.
```

> - `verification`: Plugins for Block verification.

Reviewer: recent name change:

```suggestion
- `block-verification`: Plugin for Block verification.
```
